iceoryx_hoofs/test/moduletests/test_posix_ipc_unix_domain_sockets.cpp

```diff
@@ -374,6 +414,258 @@ TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfMultipleMessagesWithSendA
+        [&](auto& msg) { return client.timedSend(msg, 1_ms); },
+        [&](auto& msg) { return server.receive(msg); });
+}
+
+TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfNonEmptyMessageWithTimedSendAndTimedReceive128Msg)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "e48b9dbb-6f9c-4e78-a053-9129f6604a0b");
+    successfulSendAndReceiveMsg<message_t, message128_t>(
+        {"it is not the sun that rises, it is hypnotoad who is opening its eyes"},
+        [&](auto& msg) { return client.timedSend(msg, 1_ms); },
+        [&](auto& msg) { return server.timedReceive(msg, 1_ms); });
+}
+
+TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfNonEmptyMessageWithSendAndTimedReceive128Msg)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "dd1ce7b1-23fe-4547-b3ad-44531f5c9d5f");
+    successfulSendAndReceiveMsg<message_t, message128_t>(
+        {"what is the most beautiful color in the world? it's hypnotoad."},
+        [&](auto& msg) { return client.send(msg); },
+        [&](auto& msg) { return server.timedReceive(msg, 1_ms); });
+}
+
+TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfEmptyMessageWithSendAndReceiveMsg)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "b5c5f2bc-b319-4b75-86c2-c44ddd5f8d75");
+    successfulSendAndReceiveMsg<message_t, message_t>(
+        {""}, [&](auto& msg) { return client.send(msg); }, [&](auto& msg) { return server.receive(msg); });
+}
+
+TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfEmptyMessageWithTimedSendAndReceiveMsg)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "ccbbb0bb-f6b7-420b-9713-7642fd8f4766");
+    successfulSendAndReceiveMsg<message_t, message_t>(
+        {""}, [&](auto& msg) { return client.timedSend(msg, 1_ms); }, [&](auto& msg) { return server.receive(msg); });
+}
+
+TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfEmptyMessageWithTimedSendAndTimedReceiveMsg)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "5c076821-d02b-4ba8-9329-a8c19555229c");
+    successfulSendAndReceiveMsg<message_t, message_t>(
+        {""},
+        [&](auto& msg) { return client.timedSend(msg, 1_ms); },
+        [&](auto& msg) { return server.timedReceive(msg, 1_ms); });
+}
+
+TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfEmptyMessageWithSendAndTimedReceiveMsg)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f68cfc06-07ae-4830-9f06-0127ecb7bcd8");
+    successfulSendAndReceiveMsg<message_t, message_t>(
+        {""}, [&](auto& msg) { return client.send(msg); }, [&](auto& msg) { return server.timedReceive(msg, 1_ms); });
+}
+
+TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfFullLenghtMessageWithSendAndReceiveMsg)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "dc059e53-3d31-4ad5-93be-bbf0a1c0425d");
+    auto message = memsetMessage<message_t>('a');
+    successfulSendAndReceiveMsg<message_t, message_t>(
+        {message}, [&](auto& msg) { return client.send(msg); }, [&](auto& msg) { return server.receive(msg); });
+}
```

Comment:
I think I was a bit unclear with my `a test with a iox::string full of e.g. a` comment. It's great to have this tests with a full buffer since it is an edge case but can you add another one.

My initial intention was to have a receive buffer full of `a` and receive something small, e.g. `all glory to the hypnotoad`. The reason for this is that the whole buffer of the `iox::string` is currently set to 0. We should rethink this decision but it also means that an off by one error is easily overlooked due to the zeroed buffer. It doesn't mean that there is one currently there but the tests are also there to prevent one in a future refactoring.

What we need is basically this
```cpp
successfulSendAndReceiveMsg<message_t, message_t>(
    {"All glory to the hypnotoad"}, [&](auto& msg) { return client.send(msg); }, [&](auto& msg) {
        msg = memsetMessage<message_t>('a');
        return server.receive(msg);
    });
```
It is sufficient to have it only once and does not need to be repeated for timedSend/timedReceive.

Reply:
https://github.com/lucabart97/iceoryx/blob/641d81d534936565898e1f9a81e70b70a9754c56/iceoryx_hoofs/test/moduletests/test_posix_ipc_unix_domain_sockets.cpp#L615-L625